solidity/contracts/bridge/Wallets.sol

```diff
@@ -624,4 +624,34 @@ library Wallets {
 
         closeWallet(self, walletPubKeyHash);
     }
+
+    /// @notice Notifies about a timed out moving funds process. Terminates
+    ///         the wallet in result.
+    /// @param walletPubKeyHash 20-byte public key hash of the wallet
+    /// @dev Requirements:
+    ///      - The wallet must be in the MovingFunds state
+    ///      - The moving funds timeout must be actually exceeded
+    function notifyWalletMovingFundsTimeout(
+        BridgeState.Storage storage self,
+        bytes20 walletPubKeyHash
+    ) internal {
+        Wallet storage wallet = self.registeredWallets[walletPubKeyHash];
+
+        require(
+            wallet.state == WalletState.MovingFunds,
+            "ECDSA wallet must be in MovingFunds state"
+        );
+
+        require(
+            /* solhint-disable-next-line not-rely-on-time */
+            block.timestamp >
+                wallet.movingFundsRequestedAt + self.movingFundsTimeout,
+            "Moving funds has not timed out"
+        );
+
+        terminateWallet(self, walletPubKeyHash);
+
+        // TODO: Perform slashing of wallet operators, reward the notifier
+        //       using seized amount, and add unit tests for that.
```

Comment:
If you review https://github.com/keep-network/keep-core/pull/2942 we can do it now 😉 

It would be great to test it with a real `WalletRegistry` and `TokenStaking`. We can have a separate test file and setup for that to do not complicate tests covering corner cases but it's definitely worth having the integration unit-tested as well. We will at least see the real gas cost.

Reply:
I can review it but I prefer to implement all the slashing/seizing separately. Agreed about the unit tests, we can add them once the integration is ready in the code.